google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPoolOptions.java

```diff
@@ -669,6 +698,49 @@ Builder setPoolMaintainerClock(Clock poolMaintainerClock) {
       return this;
     }
 
+    /**
+     * Sets whether the client should use multiplexed session or not. If set to true, the client
+     * optimises and runs multiple applicable requests concurrently on a single session. A single
+     * multiplexed session is sufficient to handle all concurrent traffic, so we don't require to
+     * provision any additional sessions.
+     *
+     * <p>When set to false, the client uses the regular session cached in the session pool for
+     * running 1 concurrent transaction per session. We require to provision sufficient sessions by
+     * making use of {@link SessionPoolOptions#minSessions} and {@link
+     * SessionPoolOptions#maxSessions} based on the traffic load. Failing to do so will result in
+     * higher latencies.
+     *
+     * @param useMultiplexedSession
+     */
+    Builder setUseMultiplexedSession(boolean useMultiplexedSession) {
+      this.useMultiplexedSession = useMultiplexedSession;
+      return this;
+    }
+
+    @VisibleForTesting
+    Builder setMultiplexedSessionMaintenanceDuration(
+        Duration multiplexedSessionMaintenanceDuration) {
+      this.multiplexedSessionMaintenanceDuration = multiplexedSessionMaintenanceDuration;
+      return this;
+    }
+
+    /**
+     * This option is only useful when {@link SessionPoolOptions#useMultiplexedSession} is set to
+     * true. If greater than zero, we wait for the said duration at time of client initialization to
+     * ensure that the multiplexed session is initialised. The default value for this is 10s.
+     *
+     * <p>If this is set to null, the client does not wait for session is initialised which means
+     * that the first few requests would see more latency since they will need to wait till the
+     * multiplexed session is initialised.
+     *
+     * @param waitForMultiplexedSession
+     * @return
+     */
+    Builder setWaitForMultiplexedSession(Duration waitForMultiplexedSession) {
```

Comment:
I think that this should somehow be combined with the `waitForMinSessions` option. It is illogical if the session pool blocks for multiplexed sessions, and not for other sessions, as it means that:
1. The first read operation will not have any additional latency.
2. The first write operations could have additional latency.

Or if we decide to keep them separate, then we should at least document this difference, so users can figure out why the first read/write transaction could get additional latency.